pkgs/applications/graphics/gnome-decoder/default.nix

```diff
@@ -0,0 +1,89 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, clang
+, libclang
+, rustPlatform
+, meson
+, ninja
+, pkg-config
+, glib
+, gtk4
+, libadwaita
+, zbar
+, sqlite
+, pipewire
+, gst_all_1
+, wrapGAppsHook4
+, appstream-glib
+, desktop-file-utils
+}:
+
+let
+  # require 'libadwaita >= 1.2.alpha'
+  libadwaita-git = libadwaita.overrideAttrs (oldAttrs: rec {
+    version = "1.2.alpha";
+    src = fetchFromGitLab {
+      domain = "gitlab.gnome.org";
+      owner = "GNOME";
+      repo = "libadwaita";
+      rev = version;
+      hash = "sha256-JMxUeIOUPp9k5pImQqWLVkQ2GHaKvopvg6ol9pvA+Bk=";
+    };
+  });
+in
+stdenv.mkDerivation rec {
+  pname = "gnome-decoder";
+  version = "0.3.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "decoder";
+    rev = "${version}";
+    sha256 = "sha256-WJIOaYSesvLmOzF1Q6o6aLr4KJanXVaNa+r+2LlpKHQ=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "sha256-RMHVrv/0q42qFUXyd5BSymzx+BxiyqTX0Jzmxnlhyr4=";
+  };
+
+  nativeBuildInputs = [
+    clang
```

Comment:
Do we want to use the clangStdenv?

Reply:
This is probably just needed for some Rust package. A comment might be nice.